app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+        }
+
+        // Sub keywords
+        $subs = explode(',', $request->get('subkws'));
+        foreach ($subs as $item) {
+            $trimmedItem = trim($item);
+            if(strlen($trimmedItem) == 0) {
+                continue;
+            }
+
+            $subKeyword = Keywords::getKeyword($trimmedItem, 2);
+
+            if(empty($subKeyword)) {
+                // Insert new keyword
+                $subKeyword = new Keywords();
+                $subKeyword->keyword = $trimmedItem;
+                $subKeyword->type = 2;
+                $subKeyword->save();
+
+                // Save post-keyword relationship
+            }
+
+            $postMetaSkw = PostMeta::getPostMeta($post->id, $subKeyword->id);
+            if(empty($postMetaSkw)) {
+                // Insert new post-keyword relationship if not exist
+                $postMetaSkw = new PostMeta();
+                $postMetaSkw->post_id = $post->id;
+                $postMetaSkw->keyword_id = $subKeyword->id;
+                $postMetaSkw->save();
+            }
+
+
+        }
+
+        return [
+            'data' => [
+                'id' => $post->id,
+                'checksum' => $calculatedChecksum
+            ],
+            'next' => action('Webapp\\PostController@getEdit', $post->id)
+        ];
+    }
+
+    public function getEdit(Request $request, $postId = 0) {
+        $this->activityLog->setPostId($postId);
+
+        if(empty($post = Post::find($postId))
+            || !$post->canViewBy($this->currentUser)
+            || ($post->site_id > 0 && Site::find($post->site_id)->status != 1) ) {
+            $this->activityLog->setInfo('Not found');
+            return '404 Not Found';
+        }
+
+        $readOnly = !$post->canEditBy($this->currentUser);
+
+        // Lock detect
+        if(!($lockStatus = $post->isLock($this->currentSession)) && !$readOnly) {
+            $post->lock($this->currentSession);
+        }
+
+        $categoriesSet = [];
+
+        foreach(Category::all() as $category) {
+            if(!isset($categoriesSet[$category->site_id])) {
+                $categoriesSet[$category->site_id] = [];
+            }
+            $categoriesSet[$category->site_id][] = $category;
+        }
+
+        $post->site = Site::find($post->site_id);
+
+        $pmt = new PostMeta();
+        $pmt_data = $pmt->getPostMetaByPostId($post->id);
+        $subkws = '';
+        $mainkw = '';
+        foreach ($pmt_data as $value) {
+            if($value->type == 2) {
+                $subkws .= $value->keyword . ',';
+            } elseif ($value->type == 1) {
+                $mainkw .= $value->keyword . ',';
+            }
+        }
+        $action_delete = $this->currentUser->is(User::ADMIN) ? 'Webapp\PostController@getDelete' : 'Webapp\PostController@getWriterDelete';
+
+        $clientChecksum = $post->getLastClientChecksum();
+        $recoverChecksum = null;
+
+        if(!empty($checksum) && $post->getChecksum() != $clientChecksum) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Review the data-flow - use of uninitialized value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS836)